sdk/servicebus/service-bus/test/internal/streamingReceiver.spec.ts

```diff
@@ -10,10 +10,211 @@ import { ReceiveOptions } from "../../src/core/messageReceiver";
 import { OperationOptions } from "../../src";
 import { StreamingReceiver } from "../../src/core/streamingReceiver";
 import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import sinon from "sinon";
 chai.use(chaiAsPromised);
 const assert = chai.assert;
 
 describe("StreamingReceiver unit tests", () => {
+  describe("receive(), close() and stopReceivingMessages() interactions", () => {
+    it("off by default", async () => {
+      const streamingReceiver = new StreamingReceiver(createClientEntityContextForTests());
+      assert.isFalse(streamingReceiver.isReceivingMessages);
+
+      // init() is considered the start of the receive operation (from StreamingReceiver's point of
+      // view)
+      await streamingReceiver.init(true);
+      assert.isTrue(streamingReceiver.isReceivingMessages);
+    });
+
+    it("isReceivingMessages set to true if subscribe() is called", async () => {
+      const streamingReceiver = new StreamingReceiver(createClientEntityContextForTests(), {
+        maxConcurrentCalls: 101
+      });
+
+      await streamingReceiver.init(true);
+
+      // 'init' is always the predecessor to calling receive so we consider that to be the
+      // start of the receive call.
+      assert.isTrue(
+        streamingReceiver.isReceivingMessages,
+        "receive() sets the isReceivingMessages flag immediately to avoid race conditions"
+      );
+
+      streamingReceiver.subscribe(
+        async (_msg) => {},
+        async (_err) => {}
+      );
+
+      assert.equal(
+        streamingReceiver["_receiver"]!.credit,
+        101,
+        "Credits are added when receive() is called"
+      );
+
+      assert.isTrue(
+        streamingReceiver.isReceivingMessages,
+        "receive() sets the isReceivingMessages flag immediately to avoid race conditions"
+      );
+
+      assert.equal(streamingReceiver["_receiver"]?.credit, 101);
```

Comment:
Looks like this check is already done at lines 48-49

Reply:
Indeed! Removing!